### PR TITLE
opkg-keyrings: mark architecture as all

### PR DESCRIPTION
**Pull request by pratheekshasn**
### Summary of Changes

The PR changes the architecture of the opkg-keyrings IPK so that it is marked with the all-arch.

### Justification

[#AB2346871](https://dev.azure.com/ni/DevCentral/_workitems/edit/2346871)

NILRT deployments which are very old might conceivably outlive the lifetimes of their default opkg signing keys. Eg. a NILRT 20.0 deployment with only the 2019 signing key may be deployed in 2029. NI doesn't provide an appealing user interface to manipulate the opkg keystore. But we should give customers a pathway to upgrade their signing keys which is more easy than hacking newer OE feeds onto their device.

As part of this, there will be some changes made to opkg-keyrings IPK. This PR is a follow-up to [!739](https://github.com/ni/meta-nilrt/pull/739).
The current PR marks the package as all-arch, since the IPK is marked as being architecture-specific by default, even though it contains entirely cross-platform content like the keyrings themselves.

### Testing
Tested that the IPK is built under deploy/ipk/all.
image

* [x] I have built the core package feed with this PR in place. (`bitbake packagefeed-ni-core`)

### Procedure

* [x] I certify that the contents of this pull request complies with the [Developer Certificate of Origin](https://github.com/ni/nilrt/blob/HEAD/docs/CONTRIBUTING.md#developer-certificate-of-origin-dco).

Signed-off by: Pratheeksha S N <pratheeksha.s.n@ni.com>

### CODE REVIEW

File: meta/recipes-devtools/opkg/opkg-keyrings_1.0.bb [hunk 1]
```diff
@@ -2,6 +2,8 @@ SUMMARY = "Keyrings for verifying opkg packages and feeds"
 LICENSE = "MIT"
 LIC_FILES_CHKSUM = "file://${COREBASE}/meta/COPYING.MIT;md5=3da9cfbcb788c80a0384361b4de20420"
 
+PACKAGE_ARCH = "all"
```
Comment (chaitu236) [inline meta/recipes-devtools/opkg/opkg-keyrings_1.0.bb]: I notice some recipes using `inherit allarch` which also seems related. Can you check if that needs to be added in addition to this or in place of this?

**Review by chaitu236 — CHANGES_REQUESTED** [review]
> The PR changes the version of the opkg-keyrings IPK so that it is shipped with the distro version.

The summary doesn't match changes here. Please fix

> This PR is a follow-up to !739

Can you add a link to the PR as just a number 739 doesn't tell us which repo this PR belongs to.

File: meta/recipes-devtools/opkg/opkg-keyrings_1.0.bb (hunk 1 above)
Comment (pratheekshasn) [reply meta/recipes-devtools/opkg/opkg-keyrings_1.0.bb]: You're right, I just need to add `inherit allarch` in place of this.

**Comment by pratheekshasn** [comment]
I have fixed the summary and the description.

**Comment by chaitu236** [comment]
> I have fixed the summary and the description.

Please also describe why the change is being made in the 2nd line of commit description.. something like

`inherit allarch as package contains entirely cross-platform content.`

And then this can be sent upstream. Once it is merged in upstream, we'll get the change on next upstream merge (or we can cherry-pick from upstream's master branch). But if you need this commit to be merged right away in our repos without waiting for upstream, let me know.

**Review by chaitu236 — APPROVED** [review]

**Comment by pratheekshasn** [comment]
I have fixed the description in the commit as well.

**Comment by pratheekshasn** [comment]
I have sent the patch upstream, but they are asking for some testing and further work. Can we merge this for now into our repo and then later cherry-pick it from upstream once it's in?

@chaitu236, @amstewart

**Comment by amstewart** [comment]
@pratheekshasn I cherry-picked your commit to `nilrt/master/scarthgap` as [this commit](https://github.com/ni/openembedded-core/commit/d1308e9f54be0b95ea2b893f8a871b9fff900d53). You can close this PR.

**Pull request closed without merging**